utils/resolver_test.go

```diff
@@ -25,3 +28,64 @@ func TestSnakify(t *testing.T) {
 		}
 	}
 }
+
+type testResolvableChild struct {
+	foo int
+}
+
+type testResolvableParent struct {
+	child *testResolvableChild
+	bar   string
+}
+
+func (p *testResolvableParent) Resolve(key string) interface{} {
+	switch key {
+	case "child":
+		return p.child
+	case "bar":
+		return p.bar
+	}
+	return fmt.Errorf("no such field on parent")
+}
+
+func (p *testResolvableParent) Default() interface{} {
```

Comment:
I've now convinced myself that Default is the right name for this afterall since it will usually return something that isn't a primitive.